src/visual/dataColor.js

```diff
@@ -74,6 +77,25 @@ export default function (seriesType) {
                     // Set data all color for legend
                     dataAll.setItemVisual(rawIdx, 'color', singleDataColor);
                 }
+
+                if (!singleDataBorderColor) {
+                    var borderColor = itemModel.get('itemStyle.borderColor')
```

Comment:
For some historical reason, we have both `'barBorderColor'` and `'borderColor'`.